components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -775,18 +776,43 @@ IJsonSerializer jsonSerializer
         }
 
         /// <inheritdoc/>
-        public async Task<PublishedNodesEntryModel> GetConfiguredNodesOnEndpointAsync(
+        public async Task<List<OpcNodeOnEndpointModel>> GetConfiguredNodesOnEndpointAsync(
             PublishedNodesEntryModel request,
             CancellationToken ct = default) {
 
             _logger.Information("{nameof} method triggered", nameof(GetConfiguredNodesOnEndpointAsync));
+            var sw = Stopwatch.StartNew();
             await _lockConfig.WaitAsync(ct).ConfigureAwait(false);
+            List<OpcNodeOnEndpointModel> response = new List<OpcNodeOnEndpointModel>();
             try {
-                throw new MethodCallStatusException((int)HttpStatusCode.NotImplemented, "Not Implemented");
+                var nodeFound = false;
+                if (!_publishedNodesEntries.Any()) {
```

Comment:
Let's get rid of this if and just do the `foreach` loop that is present in the `else` block. If `_publishedNodesEntries` list is empty `foreach` loop will detect it and we will not need to have to throw the very same `MethodCallStatusException((int)HttpStatusCode.NotFound, "Node not found in endpoint.")` exception in two places.

Reply:
removed